lua/nvim-tree/explorer/explore.lua

```diff
@@ -40,22 +40,24 @@ local function populate_children(handle, cwd, node, git_status, parent)
     ---@type uv.fs_stat.result|nil
     local stat = vim.loop.fs_stat(abs)
     local filter_reason = parent.filters:should_filter_as_reason(abs, stat, filter_status)
-    if filter_reason == FILTER_REASON.none and not nodes_by_path[abs] and Watcher.is_fs_event_capable(abs) then
-      local child = nil
-      if t == "directory" and vim.loop.fs_access(abs, "R") then
-        child = builders.folder(node, abs, name, stat)
-      elseif t == "file" then
-        child = builders.file(node, abs, name, stat)
-      elseif t == "link" then
-        local link = builders.link(node, abs, name, stat)
-        if link.link_to ~= nil then
-          child = link
+    if filter_reason == FILTER_REASON.none and not nodes_by_path[abs] then
+      if Watcher.is_fs_event_capable(abs) then
```

Comment:
This will be a change in behaviour for all users; unfortunately we can't do that.